server/graphql/v2/mutation/EmojiReactionMutations.js

```diff
@@ -123,37 +123,23 @@ const addReactionToCommentOrUpdate = async (id, req, emoji, identifierType) => {
 
 const removeReactionFromCommentOrUpdate = async (id, remoteUser, emoji, identifierType) => {
   const commentOrUpdateId = idDecode(id, identifierType);
-  let reaction;
-  if (identifierType === IDENTIFIER_TYPES.COMMENT) {
-    reaction = await models.EmojiReaction.findOne({
-      where: {
-        CommentId: commentOrUpdateId,
-        emoji,
-      },
-    });
-  } else {
-    reaction = await models.EmojiReaction.findOne({
-      where: {
-        UpdateId: commentOrUpdateId,
-        emoji,
-      },
-    });
-  }
-
-  if (!reaction) {
-    throw new NotFound(`This reaction does not exist or has been deleted.`);
-  }
+  const idColumn = identifierType === IDENTIFIER_TYPES.COMMENT ? 'CommentId' : 'UpdateId';
+  const emojiRemoved = await models.EmojiReaction.destroy({
+    where: {
+      [idColumn]: commentOrUpdateId,
+      UserId: remoteUser.id,
+      emoji,
+    },
+  });
 
-  // Check permissions
-  if (!remoteUser.isAdmin(reaction.FromCollectiveId)) {
-    throw new Forbidden();
+  if (!emojiRemoved) {
+    throw new NotFound(`This reaction does not exist or has been deleted or you do not have permission to change it.`);
   }
 
-  await reaction.destroy();
   if (identifierType === IDENTIFIER_TYPES.COMMENT) {
-    return { comment: models.Comment.findByPk(reaction.CommentId), update: null };
+    return { comment: models.Comment.findByPk(commentOrUpdateId), update: null };
```

Comment:
How does this work exactly? Shouldn't we await for the promise to resolve?
```suggestion
    return { comment: await models.Comment.findByPk(commentOrUpdateId), update: null };
```

Reply:
I think it works because according to the doc; https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#rewriting_a_promise_chain_with_an_async_function

```
The return value of an async function is implicitly wrapped in Promise.resolve - if it's not already a promise itself (as in this example).
```

So my understanding is that for the return value we don't need the `await`. Am I missing something? 🤔

Reply:
You're not returning a promise here, you're returning an object.

Reply:
@kewitz : Yes you are right. I was confused on this; we are returning an object; but I need to figure out why it was working without the await. Let me think more on it. 🤔

Reply:
@kewitz : My thinking is we are seeing this as working because here we aren't waiting for the value in the front-end but using the optimistic response. 

https://github.com/opencollective/opencollective-frontend/blob/6d05f55b84c378f7d716c8033fbba0c9a4da6430/components/conversations/EmojiReactionPicker.js#L75

So I agree that here we should have an `await`.

Reply:
I think it worked because GraphQL will await each `resolve` of the `EmojiReactionsResponse` object's fields. 

I also prefer the `await` approach though.